subsys/shell/modules/kernel_service.c

```diff
@@ -329,6 +358,9 @@ SHELL_STATIC_SUBCMD_SET_CREATE(sub_kernel,
 		defined(CONFIG_THREAD_MONITOR)
 	SHELL_CMD(stacks, NULL, "List threads stack usage.", cmd_kernel_stacks),
 	SHELL_CMD(threads, NULL, "List kernel threads.", cmd_kernel_threads),
+#endif
+#if defined(CONFIG_SYS_HEAP_RUNTIME_STATS) && (CONFIG_HEAP_MEM_POOL_SIZE > 0)
```

Comment:
Did you could consider `SHELL_SUBCMD_COND_ADD` instead of this `ifdef`?

Reply:
I was unaware of these conditional macros. I had a look, and probably the one to be used would be `SHELL_COND_CMD`. However, we have to check that both `CONFIG_SYS_HEAP_RUNTIME_STATS` is enabled and `CONFIG_HEAP_MEM_POOL_SIZE` is greater than zero. This doesn't seem to work directly as a condition for `SHELL_COND_CMD`, but I would have to define an intermediate compile time flag for this.

Another downside of `SHELL_COND_CMD` seems to be that even if the condition is not true, the handler function needs to be present, otherwise compilation fails. The implementation needs to be flagged, because `_system_heap` does not exist if `CONFIG_HEAP_MEM_POOL_SIZE` is zero.

I propose that we keep the implementation as it is.